Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py

```diff
@@ -1750,15 +1750,17 @@ def build_search_human_readable(args, parsed_search_results):
 
 def update_headers_from_field_names(search_result, chosen_fields):
     headers = []
-    search_result_keys = search_result[0].keys()
+    result_keys = set()
+    for search_result_keys in search_result:
+        result_keys = set(search_result_keys.keys())
```

Comment:
we override `result_keys` in every iteration, shouldnt we `add` on each instead?